primitives/runtime/src/transaction_validity.rs

```diff
@@ -50,6 +50,13 @@ pub enum InvalidTransaction {
 	///
 	/// The transaction might be valid, but there are not enough resources left in the current block.
 	ExhaustsResources,
+	/// An extrinsic with a Mandatory dispatch resulted in Error. This is indicative of either a
+	/// malicious validator or a buggy `provide_inherent`. In any case, it can result in dangerously
+	/// overweight blocks and therefore if found, invalidates the block.
+	BadMandatory,
```

Comment:
If you move both below `Custom` we don't need any real upgrade logic, besides having all validators updated before enabling the runtime upgrade.

Reply:
What is the update logic needed, if we let these stay there? I can only guess since this data is passed back and forth to the runtime, the previous encodings are now invalid

Would be very good to document such cases: do not change ordering. I think the same goes for the call enums as well.